### ISSUE TITLE
fasterRCNN rpn cls pred的问题

### ISSUE THREAD

**Issue by littletomatodonkey** [PaddlePaddle/models#1559]
在mdoels中fasterRCNN复现模型中，计算`rpn cls score`的代码为:[链接](https://github.com/PaddlePaddle/models/blob/cbe656e04e9445699eea6612662995f1f4593a6c/fluid/PaddleCV/faster_rcnn/models/model_builder.py#L119)，这里卷积之后没有sigmoid激活函数进行处理，但是在后面使用`rpn_target_assign`调用rpn_cls_score的时候，我看paddle的[文档](http://www.paddlepaddle.org/documentation/docs/zh/1.2/api_cn/layers_cn.html#rpn-target-assign)里写了输入的`rpn cls score`需要是sigmoid logits，我想问一下现在的逻辑是在`rpn_target_assign`函数中对rpn conv的输出进行sigmoid的操作吗？我看[https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/operators/detection/rpn_target_assign_op.cc](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/operators/detection/rpn_target_assign_op.cc)里面的代码没怎么看懂。。。没找到sigmoid对应的操作。。

**Comment by guoshengCS** [comment]
sigmoid激活函数的调用在这里
https://github.com/PaddlePaddle/models/blob/cbe656e04e9445699eea6612662995f1f4593a6c/fluid/PaddleCV/faster_rcnn/models/model_builder.py#L151

**Comment by littletomatodonkey** [comment]
那个计算得到的是`rpn_cls_score_prob`变量，但是在计算rpn_loss的时候，使用的是`rpn_cls_score`，`rpn_cls_score_prob`只在[generate_proposals函数](https://github.com/PaddlePaddle/models/blob/cbe656e04e9445699eea6612662995f1f4593a6c/fluid/PaddleCV/faster_rcnn/models/model_builder.py#L160)中用到。不知道这是为什么？谢谢！

**Comment by guoshengCS** [comment]
看了下，这个rpn_cls_score的输入没有在rpn_target_assign_op里使用，只是与rpn_target_assign_op输出的index进行了gather操作， https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/fluid/layers/detection.py#L192

**Comment by littletomatodonkey** [comment]
明白了，我刚才看计算softmax loss的时候，输入的是softmax激活之后的layer，但是计算sigmoid_cross_entropy_with_logits的时候，label是logits，x是激活之前的layer就行，会在计算sigmoid_cross_entropy_with_logits的时候求sigmoid，所以在计算fasterRCNN loss的时候不需要首先计算sigmoid，谢谢啦。
